header-rewrite-filter/header_processor.cc

```diff
@@ -322,31 +331,54 @@ namespace HeaderRewriteFilter {
                     if (start + 4 == operation_expression.end()) {
                         throw std::out_of_range("unexpected end of expression");
                     }
+
                     std::string string_to_compare(*(start + 4));
-                    matcher_ = [string_to_compare](std::string source) { return source.length() > 0 && source.compare(string_to_compare) == 0; };
+                    const absl::Status parse_status = stringToCompareSetup(string_to_compare);
+                    if (parse_status != absl::OkStatus()) {
+                        return parse_status;
+                    }
+
+                    matcher_ = [](std::string source, std::string string_to_compare) { return source.length() > 0 && source.compare(string_to_compare) == 0; };
                     break;
                 }
                 case Utility::MatchType::Prefix:
                 {
                     if (start + 4 == operation_expression.end()) {
                         throw std::out_of_range("unexpected end of expression");
                     }
+
                     std::string string_to_compare(*(start + 4));
-                    matcher_ = [string_to_compare](std::string source) { return source.length() > 0 && string_to_compare.find(source.c_str(), 0, string_to_compare.size()) == 0; };
+                    const absl::Status parse_status = stringToCompareSetup(string_to_compare);
+                    if (parse_status != absl::OkStatus()) {
+                        return parse_status;
+                    }
+
+                    matcher_ = [](std::string source, std::string string_to_compare) { return source.length() > 0 && string_to_compare.find(source.c_str(), 0, string_to_compare.size()) == 0; };
```

Comment:
I still think the params `source` and `string_to_compare` should be fine as `const&` or `string_view`. 
They're evaluated when the function is called, and the function doesn't keep them around, references should be totally fine memory-wise.
But I know it had weird errors before, so not a huge deal.

Reply:
Modified them to be `const&`.